libyang 1.0.73

li-pingmao opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**
-  Current libyang 0.16 supports YANG 1.0,  change libyang Makefile to pull libyang 1.0.73 from libyang github source and build libyang debian package
-  Build libyang python2 and python3 debian package for Sonic CLI 

**- How I did it**
- libyang Makefile changes
- patch for debian packages for libyang python2 and python3 debian packages

**- How to verify it**
Build and install libyang, libyang python2 and libyang python3 debian packages

To build:
   make target/debs/stretch/libyang_1.0.70_amd64.deb

to install libyang python2:

sudo dpkg -i target/debs/stretch/libyang_1.0.73_amd64.deb
sudo dpkg -i target/debs/stretch/libyang-cpp_1.0.73_amd64.deb
sudo dpkg -i target/debs/stretch/python2-yang_1.0.73_amd64.deb

to install libyang python3:

sudo dpkg -i target/debs/stretch/python2-yang_1.0.73_amd64.deb

zhenggen-xu commented on rules/libyang.mk:
```diff
@@ -1,23 +1,31 @@
 
-LIBYANG = libyang$(LIBYANG_VERSION_BASE)_$(LIBYANG_VERSION)-$(LIBYANG_SUBVERSION)_$(CONFIGURED_ARCH).deb
+#LIBYANG = libyang$(LIBYANG_VERSION_BASE)_$(LIBYANG_VERSION)-$(LIBYANG_SUBVERSION)_$(CONFIGURED_ARCH).deb
+LIBYANG = libyang_$(LIBYANG_VERSION)_$(CONFIGURED_ARCH).deb
```
Any reason we want to change the package name?

li-pingmao replied:
To match the built libyang package name.   Either change the package name here or change the makefile of the libyang for the debian package name

zhenggen-xu replied:
We can keep the libang default package name,  and let the dependent party to use the new name.

zhenggen-xu commented on rules/libyang.mk:
```diff
@@ -1,23 +1,31 @@
-LIBYANG_DEV = libyang-dev_$(LIBYANG_VERSION)-$(LIBYANG_SUBVERSION)_$(CONFIGURED_ARCH).deb
-$(eval $(call add_derived_package,$(LIBYANG),$(LIBYANG_DEV)))
+#LIBYANG_DEV = libyang-dev_$(LIBYANG_VERSION)-$(LIBYANG_SUBVERSION)_$(CONFIGURED_ARCH).deb
+LIBYANG_DEV = libyang-dev_$(LIBYANG_VERSION)_$(CONFIGURED_ARCH).deb
```
same question. Reason for changing the package name?

li-pingmao replied:
Use the Debian package names built from libyang GitHub

zhenggen-xu commented on src/libyang/Makefile:
```diff
@@ -3,19 +3,27 @@ SHELL = /bin/bash
 
 MAIN_TARGET = $(LIBYANG)
-DERIVED_TARGETS = $(LIBYANG_DEV) $(LIBYANG_DBG)
+DERIVED_TARGETS = $(LIBYANG_DEV) $(LIBYANG_DBG) $(LIBYANG_PY2) $(LIBYANG_CPP)
```
We don't need LIBYANG_PY3 here?

zhenggen-xu commented on src/libyang/patch/libyang.patch:
```diff
@@ -0,0 +1,133 @@
+ %endif
+ %endif
+ 
++%if 0%{?suse_version} + 0%{?fedora} > 0
```
It looks like we don't need this if/else here.

zhenggen-xu commented on src/sonic-yang-mgmt/yang-models/sonic-acl.yang:
```diff
@@ -1,4 +1,5 @@
 module sonic-acl {
+        yang-version 1.1;
```
May not be in this PR, just a note: should we add 1.1 for all yang models?

li-pingmao committed:
libyang 1.0.73
- build libyang1.0.73 debian pacakge from libyang github source
- build libyang python2 and python3 debian packages

li-pingmao committed:
Address review feedback
